### PR TITLE
Add support for short repository property and make it default

### PR DESCRIPTION
**Pull request by zdroid**
As requested in bevry/editions#16. Please review this to make sure there are no issues. About the README changes, I just ran the compile script (`npm run-script compile`).

/cc @balupton

### CODE REVIEW

File: source/lib/projectz.js [hunk 1]
```diff
@@ -386,8 +386,9 @@ class Projectz {
 
 		// Extract repository information
 		/* eslint no-magic-numbers: 0 */
-		let repo = this.mergedPackageData.repository.url || null
-		const githubMatch = (this.mergedPackageData.repository.url || this.mergedPackageData.homepage).match(/github\.com\/(.+?)(?:\.git|\/)?$/)
+		let repo = this.mergedPackageData.repository || null
+		const githubMatch = (this.mergedPackageData.repository.url || this.mergedPackageData.homepage).match(/github\.com\/(.+?)(?:\.git|\/)?$/) ||
+			(this.mergedPackageData.repository.match(/^(.+\/.+)$/) && !this.mergedPackageData.repository.match(/^(.+:.+\/.+)$/))
```
Comment (zdroid) [inline source/lib/projectz.js]: Couldn't do this the other way. If you have a better idea how to match `user/repo` but not `(bitbucket|gist|gitlab):user/repo`, please tell me.

Comment (balupton) [inline source/lib/projectz.js]: This should do it:

```
const githubMatch = this.mergedPackageData.repository.match(/^(?:github:)?(.+\/.+)$/) || (this.mergedPackageData.repository.url || this.mergedPackageData.homepage).match(/github\.com\/(.+?)(?:\.git|\/)?$/)
```

The `?:` means non capturing group, so whatever is in the `?:` group doesn't affect the match.

File: source/lib/projectz.js [hunk 2]
```diff
@@ -414,10 +415,7 @@ class Projectz {
 			// Fallback fields
 			extendr.defaults(this.mergedPackageData, {
 				// Fallback repository field by use of repo
-				repository: {
-					type: 'git',
-					url: `https://github.com/${githubSlug}.git`
-				},
+				repository: `${githubSlug}`,
```
Comment (balupton) [inline source/lib/projectz.js]: `repository: githubSlug,` is fine here

File: source/lib/projectz.js (hunk 1 above)
Comment (balupton) [inline source/lib/projectz.js]: This should do it:

``` javascript
const githubMatch = this.mergedPackageData.repository.match(/^(?:github:)?(.+\/.+)$/)
  || (this.mergedPackageData.repository.url || this.mergedPackageData.homepage).match(/github\.com\/(.+?)(?:\.git|\/)?$/
```

The `?:` means non capturing group, so whatever is in the `?:` group doesn't affect the match.

Comment (balupton) [inline source/lib/projectz.js]: This should do it:

``` javascript
const githubMatch = this.mergedPackageData.repository.match(/^(?:github:)?(.+\/.+)$/)
  || (this.mergedPackageData.repository.url || this.mergedPackageData.homepage).match(/github\.com\/(.+?)(?:\.git|\/)?$/
```

The `?:` means non capturing group, so whatever is in the `?:` group doesn't affect the match.

File: source/lib/projectz.js (hunk 2 above)
Comment (balupton) [inline source/lib/projectz.js]: `repository: githubSlug,` is fine here

File: source/lib/projectz.js (hunk 1 above)
Comment (zdroid) [inline source/lib/projectz.js]: It looks like that doesn't work. Basically, all it has to do is to ignore repository names which contain `:` (e.g. `bitbucket:user/repo`). I've checked it, creating GitHub reposities with `:` in it is impossible, they're automatically converted to `-`, so the regex should work fine on that.

Comment (balupton) [inline source/lib/projectz.js]: Okay, I'll clone and take a look 👍

Comment (balupton) [inline source/lib/projectz.js]:
```
		const githubMatch = this.mergedPackageData.repository.match(/^(?:github:)?([^\/]+\/[^\/]+)$/) ||
			(this.mergedPackageData.repository.url || this.mergedPackageData.homepage).match(/github\.com\/(.+?)(?:\.git|\/)?$/)
```

Seems to do the trick. Can you verify?

Comment (balupton) [inline source/lib/projectz.js]: Actually, I think I understand what you mean now. Leave it with me. Will merge it in and adjust.

**Comment by balupton** [comment]
Released to v1.2.0 👏

**Comment by balupton** [comment]
My only concern with this is if older versions of npm doesn't support this, however I would assume it is not important for npm and node at all, so I'm not fussed.

**Comment by zdroid** [comment]
I remember using this a few years ago, back in the time of node 0.x. I guess it isn't a problem for older versions.

**Comment by balupton** [comment]
Even better!

**Pull request merged**